helm/botkube/values.yaml

```diff
@@ -292,127 +287,186 @@ config:
+          name: 'SLACK_CHANNEL'
+          bindings:
+            executors:
+              - kubectl-read-only
+            sources:
+              - k8s-events
+      # -- Slack token.
+      token: 'SLACK_API_TOKEN'
+      notification:
+        # -- Configures notification type that are sent. Possible values: `short`, `long`.
+        type: short
+
+    ## Settings for Mattermost.
+    mattermost:
+      # -- If true, enables Mattermost bot.
+      enabled: false
+      # -- User in Mattermost which belongs the specified Personal Access token.
+      botName: 'BotKube'
+      # -- The URL (including http/https schema) where Mattermost is running. e.g https://example.com:9243
+      url: 'MATTERMOST_SERVER_URL'
+      # -- Personal Access token generated by BotKube user.
+      token: 'MATTERMOST_TOKEN'
+      # -- The Mattermost Team name where BotKube is added.
+      team: 'MATTERMOST_TEAM'
+      # -- Map of configured channels. The `channels` property name is an alias for a given configuration.
+      #
+      ## Format: channels.<alias>
+      channels:
+        'default':
+          # -- The Mattermost channel name for receiving BotKube alerts.
+          # The BotKube user needs to be added to it.
+          name: 'MATTERMOST_CHANNEL'
+          bindings:
+            executors:
+              - kubectl-read-only
+            sources:
+              - k8s-events
+      notification:
+        # -- Configures notification type that are sent. Possible values: `short`, `long`.
+        type: short
+
+    ## Settings for MS Teams.
+    teams:
+      # -- If true, enables MS Teams bot.
+      enabled: false
+      # -- The Bot name set while registering Bot to MS Teams.
+      botName: 'BotKube'
+      # -- The BotKube application ID generated while registering Bot to MS Teams.
+      appID: 'APPLICATION_ID'
+      # -- The BotKube application password generated while registering Bot to MS Teams.
+      appPassword: 'APPLICATION_PASSWORD'
+      # -- The path in endpoint URL provided while registering BotKube to MS Teams.
+      messagePath: "/bots/teams"
+      notification:
+        # -- Configures notification type that are sent. Possible values: `short`, `long`.
+        type: short
+      # -- The Service port for bot endpoint on BotKube container.
+      port: 3978
+
+    ## Settings for Discord.
+    discord:
+      # -- If true, enables Discord bot.
+      enabled: false
+      # -- BotKube Bot Token.
+      token: 'DISCORD_TOKEN'
+      # -- BotKube Application Client ID.
+      botID: 'DISCORD_BOT_ID'
+      # -- Map of configured channels. The `channels` property name is an alias for a given configuration.
+      #
+      ## Format: channels.<alias>
+      channels:
+        'default':
+          # -- Discord channel ID for receiving BotKube alerts.
+          # The BotKube user needs to be added to it.
+          name: 'DISCORD_CHANNEL_ID'
```

Comment:
Can we change this field to ID? It's quite confusing as we need to provide channel ID instead of name. Thanks!

Reply:
This is quite tricky, if we will change that then we will need to create a dedicated struct just for the discrod bot.
Currently, it's unified under:
```go
// ChannelBindings contains configuration bindings per channel.
type ChannelBindings struct {
	Name     string      `yaml:"name"`
	Bindings BotBindings `yaml:"bindings"`
}
```
What about changing that to the `ID` for all? I know that some uses name and some ID, but IMO in case of "unification" we can go with `ID`. We will need to make a lot of such "unification" later, if we want to have it more extensible. 

The other idea is to have 
```go
// ChannelBindings contains configuration bindings per channel.
type ChannelBindings struct {
	ID       string      `yaml:"id"`
	Name     string      `yaml:"name"`
	Bindings BotBindings `yaml:"bindings"`
}
```

Where in config we can have `channel[*].id` or `channel[*].name` - we validate that only one was set and we use the one that is available. 

WDYT?

Reply:
I'm not sure if that's a good unification, if some of the comm platforms require channel name (like Slack), and some ID (like Discord). We cannot unify that, as these are two different things - for example, Slack channel has Conversation ID, and also a name. 

Even a single struct with both ID and Name is confusing. I understand the problem, but I think the users' point of view is more important than ours (developer/maintainer one).

I would go with two separate structs to make it clearer to configure: e.g. `ChannelBindingsByName` and `ChannelBindingsByID` or whatever you like 🙂 Is that fine for you?

Reply:
>I'm not sure if that's a good unification, if some of the comm platforms require channel name (like Slack), and some ID (like Discord). We cannot unify that, as these are two different things - for example, Slack channel has Conversation ID, and also a name.

I know that, that the thing, we can say what ID means in our docs. 

Personally, I don't agree that it's necessary to have both ID and name, but let's agree to disagree - I will add a dedicated struct for that.